Modules/ExampleInverter/src/ExampleInverter.cpp

```diff
@@ -0,0 +1,48 @@
+#include "ExampleInverter.h"
+
+#include <mitkImageToItk.h>
+#include <mitkITKImageImport.h>
+#include "mitkLogMacros.h"
+
+namespace captk {
+
+ExampleInverter::ExampleInverter()
+{
+	// any setup required goes here
+}
+
+void ExampleInverter::SetInput(mitk::Image::Pointer inImagePtr)
+{
+	this->m_InputPtr = inImagePtr;
+}
+
+void ExampleInverter::Update()
+{
+	// Read as 3D double image (hardcoded) for now (anything else will crash).
+	// Later we can use predicates and/or AccessByItk to help with this.
+	using TPixel = typename short;
+	const unsigned int VImageDimension = 3;
+	typedef itk::Image<TPixel, VImageDimension> itkImageType;
+	typedef itk::InvertIntensityImageFilter<itkImageType, itkImageType> FilterType;
+
+	// There are a few ways to convert from MITK to ITK...
+	auto itkInImage = mitk::ImageToItkImage<TPixel, VImageDimension>(this->m_InputPtr.GetPointer());
```

Comment:
also you should probably change this to casting, because it will crash if you try it with an image for which the pixel type does not match.

void mitk::CastToItkImage(mitkImage,  
                    itk::Image<...>)